### PR TITLE
Create DashboardHeader component, update groups_.new route to use it

### PR DESCRIPTION
**Pull request by willbeaumont**
Issue: #42 

## Summary (1-2 sentences)

This PR simplifies dashboard development by abstracting styling for h1 header components. The new component is applied to the `groups/new` page.

## Details (reason and description of the changes)

Having a reusable component will make it easier to update dashboard header styles if they need to change in the future. We can now update in one location to update multiple pages. The style settings are the closest available default tailwindcss styles that match the figma design doc.

### Implementation details (how was this change implemented?) 

This change is dependent on @andrelandgraf's Implement loading indications (login, signup), improve layout (groups/new) PR.

### CODE REVIEW

**Comment by willbeaumont** [comment]
First PR here, please let me know if I did anything annoying or should have waited for the dependent PR to clear

File: app/components/ui/containers.tsx [hunk 1]
```diff
@@ -1,3 +1,7 @@
 export function Card({ children }: { children: React.ReactNode }) {
   return <section className="w-full bg-white shadow-lg rounded-lg px-2 py-6 lg:p-16">{children}</section>;
 }
+
+export function DashboardHeader({ children }: { children: React.ReactNode }) {
```
Comment (andrelandgraf) [inline app/components/ui/containers.tsx:5]: Nice! Thanks for adding the header!

I think we could move the header into a new file called `headers` in `components/ui`. This file is more meant for cards and other containers (styled divs).

File: app/routes/groups_.new.tsx [hunk 2]
```diff
@@ -1,11 +1,11 @@
-import { Card } from '~/components/ui/containers';
+import { Card, DashboardHeader } from '~/components/ui/containers';
 
 export default function GroupNew() {
   return (
     <div className="grow py-20 lg:py-40 bg-secondary w-full flex flex-col items-center justify-center">
       <div className="flex w-full items-center justify-center flex-col">
         <div className="max-w-[900px] w-full flex flex-col gap-3">
-          <h1 className="text-black font-bold text-4xl">Create New Group</h1>
+          <DashboardHeader>Create New Group</DashboardHeader>
```
Comment (andrelandgraf) [inline app/routes/groups_.new.tsx:8]: NIT: Maybe just `H1`? `DashboardHeader` also works, though!

File: app/routes/groups_.new.tsx [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
           <h2 className="text-black font-bold text-2xl">Your Community Starts Here</h2>
```
Comment (andrelandgraf) [inline app/routes/groups_.new.tsx:9]: You could also create a component for `H2` following the same pattern and move it in the same file!

**Review by andrelandgraf — COMMENTED** [review]
Thanks for the great PR and PR description. Added some NIT comments, otherwise LGTM!

**Comment by andrelandgraf** [comment]
Interesting, when I merged my PR and deleted the associated branch, it automatically closed this PR. I think if you just point your PR to merge into `main`, it should be fine? :)

File: app/routes/groups_.new.tsx (hunk 2 above)
Comment (willbeaumont) [reply app/routes/groups_.new.tsx:8]: That'd be easy, is it common to mirror common tags and use caps?

Comment (andrelandgraf) [reply app/routes/groups_.new.tsx:8]: Definitely seen it before! Mostly for `button` -> `Button` and headings `h1` -> `H1`. It's a nice pattern as the component name clearly communicates that it's a wrapper around an HTML element that adds some app-specific styling or functionality!

**Pull request closed without merging**